### PR TITLE
Allow filtering on repository charts list

### LINKED ISSUE #46: Allow filtering repository charts
Inside repository charts list, it can be long. Filtering would be nice
Should be a quick UI filter without re-querying the backend.

### PR DESCRIPTION
**Pull request by hdm23061993**
Fixes #46
The current fix supports filtering on 'Chart Name' column only

### Show all
image

### Filter on chart name
image

image

### CODE REVIEW

File: pkg/dashboard/static/actions.js [hunk 1]
```diff
@@ -337,4 +337,21 @@ $("#btnRollback").click(function () {
...
+    
+    $(".charts li").each(function(){
+        var chartNameElem = this.firstElementChild
+        var chartName = $(chartNameElem).text().toLowerCase()
+        if(chartName.indexOf(val) != -1) {
+            $(this).show()
+        }
+    })
+})
+window.onload = function() {
```
Comment (undera) [inline pkg/dashboard/static/actions.js:355]: What this is for? The control should be empty when the page loads.

File: pkg/dashboard/static/index.html [hunk 2]
```diff
@@ -110,20 +110,17 @@ <h2 class="mb-3">name-of-repo</h2>
             <div class="mb-5">
                 <span class="rounded rounded-1 me-2 p-1 px-2 bg-tag text-dark">URL: <span class="url fw-bold">http://somerepo/somepath</span></span>
             </div>
-
-            <!-- TODO
-            <div class="float-end">
-                   <input class="form-control form-control-sm" type="text" placeholder="Filter...">
+            <div class="row py-2 mb-3">
```
Comment (undera) [inline pkg/dashboard/static/index.html:113]: The visual of this control in TODO was chosen to be less agressive and take less space on screen. I would suggest to stick to it for consistency.

File: pkg/dashboard/static/actions.js (hunk 1 above)
Comment (hdm23061993) [reply pkg/dashboard/static/actions.js:355]: It wasn't clearing by itself on page refresh that is why I added this
Any other way to handle it?

File: pkg/dashboard/static/index.html (hunk 2 above)
Comment (hdm23061993) [reply pkg/dashboard/static/index.html:113]: that disturbs the entire layout 

image

File: pkg/dashboard/static/actions.js (hunk 1 above)
Comment (undera) [reply pkg/dashboard/static/actions.js:355]: You should place this call into this block: https://github.com/komodorio/helm-dashboard/blob/main/pkg/dashboard/static/repo.js#L23

This way, the filter will clear each time the repository is switched. Or you can place it even under `loadRepoView()` to clear it only when navigation to "Repository" section happens. Although I don't see any logic in keeping filter between different repositories.

Comment (hdm23061993) [reply pkg/dashboard/static/actions.js]: Fixed, please help review

Comment (undera) [reply pkg/dashboard/static/actions.js]: Looks good

**Comment by undera** [comment]
Amazing!

**Pull request merged**